### PR TITLE
feat: add support for Page.ResetNavigationHistory

### PR DESCRIPTION
**Pull request by jackcipher**
# Add ResetNavigationHistory functionality

## Description

This PR adds a new function `ResetNavigationHistory` to the `Page` struct, along with its `Must` variant `MustResetNavigationHistory`. This feature allows users to clear the navigation history of a page, which is particularly useful when pages are reused in a pool.

## Changes

1. Added `ResetNavigationHistory` method to `Page` struct in `page.go`
2. Added `MustResetNavigationHistory` method to `Page` struct in `must.go`
3. Added a new test function `TestPageResetNavigationHistory` in `page_test.go`

## Motivation

When pages are reused in a pool, the navigation history accumulates over time. This can lead to unexpected behavior and potentially impact performance. The ability to reset the navigation history provides more control over the page state, especially in scenarios where a clean slate is desired for each new use of a pooled page.

## Usage
Users can now reset the navigation history of a page like this:

```golang
err := page.ResetNavigationHistory()
```

Or using the Must variant:

```golang
page.MustResetNavigationHistory()
```

### CODE REVIEW

File: page_test.go
Comment (ysmood) [inline page_test.go:116]:
```suggestion
	g.Err(err)
```

**Review by ysmood — APPROVED** [review]
LGTM

**Comment by jackcipher** [comment]
I've cleaned up the commits. The changes are now based on the latest main branch. Could you please review the updated PR and consider re-approving if everything looks good?

File: .gitignore
Comment (ysmood) [inline .gitignore:10]: Please put this in your global gitignore, not in a specific project.

```suggestion
```

File: page.go
Comment (ysmood) [inline page.go:1030]:
```suggestion
				case dst <- msg.(*Message): //nolint: forcetypeassert
```

**Review by ysmood — CHANGES_REQUESTED** [review]

**Comment by jackcipher** [comment]
Hi, I've resolved the reviews.
Can you take another look at this PR and let me know if there's anything else needed before we can proceed with merging?

**Review by ysmood — APPROVED** [review]

**Pull request merged**
